core_backend/app/utils.py

```diff
@@ -194,13 +197,34 @@ def __call__(self) -> aiohttp.ClientSession:
 _HTTP_CLIENT: aiohttp.ClientSession | None = None
 
 
-def get_http_client() -> aiohttp.ClientSession:
+def get_global_http_client() -> Optional[aiohttp.ClientSession]:
+    """Return the value for the global variable _HTTP_CLIENT.
+
+    :returns:
+        The value for the global variable _HTTP_CLIENT.
     """
-    Get HTTP client
+
+    return _HTTP_CLIENT
+
+
+def set_global_http_client(http_client: HttpClient) -> None:
+    """Set the value for the global variable _HTTP_CLIENT.
+
+    :param http_client: The value to set for the global variable _HTTP_CLIENT.
     """
+
     global _HTTP_CLIENT
-    if _HTTP_CLIENT is None or _HTTP_CLIENT.closed:
+    _HTTP_CLIENT = http_client()
+
+
+def get_http_client() -> Optional[aiohttp.ClientSession]:
```

Comment:
This should not be optional right? Should always return ClientSession?

Reply:
this is `Optional` because [this line](https://github.com/IDinsight/aaq-core/blob/351aeb398aa64918bb99a1bb21c0d1fc18a4457e/core_backend/app/utils.py#L194) defines it as a union of two types.